master/buildbot/process/buildrequestdistributor.py

```diff
@@ -1181,7 +1184,9 @@ def _procesBuildRequestsActivityLoop(self):
         while 1:
             yield self.activity_lock.acquire()
 
-            self.active = self.running and (self.check_new_builds or self.check_resume_builds)
+            self.active = (not self.master.changeServicesStateRunning() and
+                           self.running and (self.check_new_builds or self.check_resume_builds))
+
```

Comment:
@BarryUnity whenever the master has finished setting up the services it will retry the builds that didn't get started during the config period.